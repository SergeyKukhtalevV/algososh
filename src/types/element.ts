import {ElementStates} from "./element-states";

export type TElement = {
  value: string;
  color: ElementStates
}

export type TNumber = {
  value: number;
  color: ElementStates
}
