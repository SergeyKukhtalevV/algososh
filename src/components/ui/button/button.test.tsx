import {Button} from "./button";
import {fireEvent, render, screen} from '@testing-library/react';
import renderer from "react-test-renderer";
import React from 'react';
import {Direction} from "../../../types/direction";

describe('ButtonComponent', () => {
  it('Button with text', ()=> {
    const button = renderer
      .create(<Button text='button' />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Button without text', ()=> {
    const button = renderer
      .create(<Button />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Disabled button', ()=> {
    const button = renderer
      .create(<Button disabled={true}/>)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Button with loader', ()=> {
    const button = renderer
      .create(<Button isLoader={true}/>)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('onClick button callback', ()=> {
    const callback = jest.fn();
    render(<Button onClick={callback}/>);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(callback).toHaveBeenCalledTimes(1);
  });
  it('Button with sorting-Ascending', ()=> {
    const button = renderer
      .create(<Button sorting={Direction.Ascending}/>)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
    it('Button with sorting-Descending', ()=> {
      const button = renderer
        .create(<Button sorting={Direction.Descending}/>)
        .toJSON();
      expect(button).toMatchSnapshot();
  });
  it('Button with linkedList-small', ()=> {
    const button = renderer
      .create(<Button linkedList={"small"}/>)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it('Button with linkedList-big', ()=> {
    const button = renderer
      .create(<Button linkedList={"big"}/>)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
});
