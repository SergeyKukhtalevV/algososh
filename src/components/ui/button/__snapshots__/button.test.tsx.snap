// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`ButtonComponent Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    button
  </p>
</button>
`;

exports[`ButtonComponent Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`ButtonComponent Button wth loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`ButtonComponent Disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
